InvenTree/templates/js/translated/order.js

```diff
@@ -2491,6 +2491,57 @@ function allocateStockToSalesOrder(order_id, line_items, options={}) {
                 },
                 value: options.shipment || null,
                 auto_fill: true,
+                secondary: {
+                    method: 'POST',
+                    title: '{% trans "Add Shipment" %}',
+                    fields: function() {
+                        var ref = null;
+
+                        // TODO: Refactor code for getting next shipment number
+                        inventreeGet(
+                            '{% url "api-so-shipment-list" %}',
+                            {
+                                order: options.order,
+                            },
+                            {
+                                async: false,
+                                success: function(results) {
+                                    // "predict" the next reference number
+                                    ref = results.length + 1;
+
+                                    var found = false;
+
+                                    while (!found) {
+
+                                        var no_match = true;
+
+                                        for (var ii = 0; ii < results.length; ii++) {
+                                            if (ref.toString() == results[ii].reference.toString()) {
+                                                no_match = false;
+                                                break;
+                                            }
+                                        }
+
+                                        if (no_match) {
+                                            break;
+                                        } else {
+                                            ref++;
+                                        }
+                                    }
+                                }
+                            }
+                        );
+
+                        var fields = salesOrderShipmentFields(options);
+
+                        fields.reference.value = ref;
+                        fields.reference.prefix = global_settings.SALESORDER_REFERENCE_PREFIX + options.reference;
+                        
+                        fields.order.hidden = false; // TODO: Hide order id field
```

Comment:
If `fields.order.hidden` is not set to `false`, the `POST` request fails as it cannot locate a div with id `#div_id_order_1`. When `fields.order.hidden` is `true`, a `<input>` tag is created with id `id_order`, rather than the expected secondary modal `<div>` tag with id `#div_id_order_1`.